straxen/plugins/led_cal/led_calibration.py

```diff
@@ -122,48 +178,130 @@ def get_records(raw_records, baseline_window):
     mask = np.where((records["record_i"] == 0) & (records["length"] == 160))[0]
     records = records[mask]
     bl = records["data"][:, baseline_window[0] : baseline_window[1]].mean(axis=1)
+    rms = records["data"][:, baseline_window[0] : baseline_window[1]].std(axis=1)
     records["data"][:, :160] = -1.0 * (records["data"][:, :160].transpose() - bl[:]).transpose()
+    records["baseline"] = bl
+    records["baseline_rms"] = rms
     return records
 
 
+def get_led_windows(
+    records,
+    default_window,
+    led_hit_extension,
+    hit_min_amplitude,
+    hit_min_height_over_noise,
+):
+    """Search for hits in the records, if a hit is found, return an interval around the hit given by
+    led_hit_extension. If no hit is found in the record, return the default window.
+
+    :param records: Array of the records to search for LED hits.
+    :param default_window: Default window to use if no LED hit is found given as a tuple (start,
+        end)
+    :param led_hit_extension: The integration window around the first hit found to use. A tuple of
+        form (samples_before, samples_after) the first LED hit.
+    :param hit_min_amplitude: Minimum amplitude of the signal to be considered a hit.
+    :param hit_min_height_over_noise: Minimum height of the signal over noise to be considered a
+        hit. :return (len(records), 2) array: Integration window for each record
+
+    """
+    hits = strax.find_hits(
+        records,
+        min_amplitude=hit_min_amplitude,
+        min_height_over_noise=hit_min_height_over_noise,
+    )
+    default_windows = np.tile(default_window, (len(records), 1))
+    return _get_led_windows(hits, default_windows, led_hit_extension)
+
+
+@numba.jit(nopython=True)
+def _get_led_windows(hits, default_windows, led_hit_extension):
+    windows = default_windows
+    last = -1
+    for hit in hits:
+        if hit["record_i"] == last:
+            continue  # If there are multiple hits in one record, ignore after the first
+
+        left = hit["left"] + led_hit_extension[0]
+        # Limit the position of the window so it stays inside the record.
+        if left < default_windows[0, 0]:
+            left = default_windows[0, 0]
+        elif left > 150 - (led_hit_extension[1] - led_hit_extension[0]):
+            left = 150 - (led_hit_extension[1] - led_hit_extension[0])
+
+        right = hit["left"] + led_hit_extension[1]
+        if right < default_windows[0, 1]:
```

Comment:
I would suggest to avoid hard-coded numbers in the plugin. Could you name it and add it as a URLConfig?

Reply:
Yes, you are right. I fixed the hard coded numbers.